src/proto/request.rs

```diff
@@ -48,15 +48,8 @@ pub struct AuthenticationRequest {
 
 #[derive(Debug)]
 pub struct TicketGrantRequest {
-    pub nonce: u32,
-    pub service_name: Name,
-    pub from: Option<SystemTime>,
-    pub until: SystemTime,
-    pub renew: Option<SystemTime>,
     pub preauth: Preauth,
-    pub etypes: Vec<EncryptionType>,
-    pub kdc_options: FlagSet<KerberosFlags>,
-    ap_req: Option<ApReqBuilder>,
+    pub(crate) req_body: KdcReqBody,
```

Comment:
This is the big change @scabrero - We can't (and shouldn't) parse the kdc-req-body until we perform the checksum over the data within. So I had to revert this to a kdcreqbody here.